csrc/ir/nodes.cpp

```diff
@@ -4532,6 +4532,38 @@ std::string MatmulOp::toInlineString(int indent_size) const {
   NVF_CHECK(false, "Tensor op can not be printed inline");
 }
 
+namespace {
+// When the contracting dimension is sharded, each device has a partial
+// matmul output and is followed by an allreduce. For loop split, this is
+// represented as an rfactored reduction. For example, for matmul, the local
+// logical domain after the rfactor is: i{DIDx}, i{M}, i{N}, r{K//d}. Unsqueeze
+// the rfactored DID axis to correctly bind with the logical domain. See
+// tests/python/test_multidevice.py/test_matmul_allreduce_loop_split
+int64_t getRFactorDeviceDimensionIndex(const TensorView* tv) {
+  // Filter out reduction dimensions so the index to `logical` directly maps to
+  // an at::Tensor axis.
+  auto logical = TensorDomain::noReductions(tv->getLogicalDomain());
+  int64_t rfactor_did_idx = -1;
+  for (auto idx : c10::irange(static_cast<int64_t>(logical.size()))) {
+    if (!logical.at(idx)->isRFactorProduct()) {
+      continue;
+    }
+
+    if (!logical.at(idx)->isDeviceDim()) {
+      continue;
+    }
```

Comment:
Any specific reason to keep these conditions separate? Is it just for readability?

Reply:
Yes, just a style preference -- composite conditions make my head hurt sometimes. 

I'll probably change it again to 

```
if (logical.at(idx)->isRFactorProduct() && logical.at(idx)->isDeviceDim()) {
  ...
}
```

It's more indentation but matches the function name more naturally.